frontend/src/features/books/EditBookForm.jsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react'
+import { useParams, useNavigate } from 'react-router-dom'
+import { useEffect, useState } from 'react'
+import { API_URL } from '../../constants'
+
+function EditBookForm() {
+    const [book, setBook] = useState(null)
+    const [title, setTitle] = useState('')
+    const [author, setAuthor] = useState('')
+    const [description, setDescription] = useState('')
+    const { id } = useParams()
+    const [, setLoading] = useState(true)
+    const [, setError] = useState(null)
+    const navigate = useNavigate()
+
+    useEffect(() => {
+        const fetchCurrentBook = async () => {
+            try {
+                const response = await fetch(`${API_URL}/${id}`)
+                if (response.ok) {
+                    const json = await response.json()
+                    setBook(json)
+                    setTitle(json.title)
+                    setAuthor(json.author)
+                    setDescription(json.description)
+                } else {
+                    throw response
+                }
+            } catch (e) {
+                console.log("an error occurred")
+            } finally {
+                setLoading(false)
```

Comment:
@nshelokar-vf, rather than using finally, we can setLoaing to true after execution of if and else block